### PR TITLE
Bump version to 0.16.2

### PR DESCRIPTION
**Pull request by mrocklin**
I would like to issue a new release of Dask.  This PR serves as short warning.  I plan to merge this in a couple hours.  

Fixes https://github.com/dask/dask/issues/3052

Is there anything stopping this release?

### CODE REVIEW

**Comment by jakirkham** [comment]
Releasing SGTM.

File: docs/release-procedure.md [hunk 1]
```diff
@@ -20,15 +20,16 @@ Release per project:
...
 
-            git clone git@github.com:conda-forge/dask-feedstock
-            cd dask-feedstock
+            git clone git@github.com:conda-forge/dask-core-feedstock
+            cd dask-core-feedstock
             conda install conda-smithy
             conda-smithy rerender
 
     *  Get sha256 hash from pypi.org
     *  Update version number and hash in recipe
```
Comment (jakirkham) [inline docs/release-procedure.md]: Maybe add `* Zero build number`

**Comment by jakirkham** [comment]
Also should the title be `0.16.2`?

File: docs/release-procedure.md (hunk 1 above)
Comment (mrocklin) [reply docs/release-procedure.md]: Whoops.  I failed to do this.  My apologies.

Comment (jakirkham) [reply docs/release-procedure.md]: No worries. More of a suggestion really. We're going to start adding contribution guidelines to the feedstocks, which should have details like this.

**Pull request merged**